lib/flowViewport/API/interfacesImp/fvpInformationInterfaceImp.cpp

```diff
@@ -0,0 +1,100 @@
+//
+// Copyright 2023 Autodesk, Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+//Local headers
+#include "fvpInformationInterfaceImp.h"
+#include "flowViewport/API/perViewportSceneIndicesData/fvpViewportInformationAndSceneIndicesPerViewportDataManager.h"
+
+#include <mutex>
+
+namespace{
+    static std::mutex _viewportInformationClient_mutex;
+}
+    
+PXR_NAMESPACE_USING_DIRECTIVE
+
+namespace FVP_NS_DEF {
+
+static InformationInterfaceImp theInterface;
+
+//Set of information clients
+static InformationClientSet _viewportInformationClients;
+
+InformationInterface& InformationInterface::Get() 
+{ 
+    return theInterface;
+}
+
+InformationInterfaceImp& InformationInterfaceImp::Get(void)
+{
+    return theInterface;
+}
+
+void InformationInterfaceImp::RegisterInformationClient(const InformationClient& client)
+{
+    std::lock_guard<std::mutex> lock(_viewportInformationClient_mutex);
+
+    InformationClient* clientNonConst = const_cast<InformationClient*>(&client);
```

Comment:
I prefer to keep a const reference as a parameter so we know the function doesn't change it. Only the storage in the std::set stores a non const, and I cannot change the std::set to "const ViewportInformation*" which I tried as I don't want SceneIndexAdded or SceneIndexRemoved to be const functions as a 3rd party using it as a callback could modify things in this function.

Reply:
This feels very awkward in terms of ownership.  We shouldn't be storing a pointer to something that is conceptually passed by value, because the argument object may disappear if it goes out of scope.  I think this should be changed.

Reply:
I changed it to storing a copy of the class since the class is not heavy and the size of the std set is the number of hydra viewports in maya.

Reply:
Looks like we're still storing pointers now though, no? Am I missing something?